src/category_theory/endomorphism.lean

```diff
@@ -57,9 +57,7 @@ end End
 
 lemma is_unit_iff_is_iso {C : Type u} [category.{v} C] {X : C} (f : End X) :
   is_unit (f : End X) ↔ is_iso f :=
-⟨λ h, { out := ⟨h.unit.inv,
-  ⟨by { convert h.unit.inv_val, exact h.unit_spec.symm, },
-    by { convert h.unit.val_inv, exact h.unit_spec.symm, }⟩⟩ },
+⟨λ h, { out := ⟨h.unit.inv, ⟨h.unit.inv_val, h.unit.val_inv⟩⟩ },
   λ h, by exactI ⟨⟨f, inv f, by simp, by simp⟩, rfl⟩⟩
```

Comment:
You can avoid using `classical.choice` here with:
```suggestion
⟨λ h, { out := by { obtain ⟨u, rfl⟩ := h, exact ⟨u.inv, ⟨u.inv_val, u.val_inv⟩⟩ } },
  λ h, let ⟨f_inv, hf, hf'⟩ := h.out in ⟨⟨f, f_inv, hf', hf⟩, rfl⟩⟩
```

Reply:
Not sure this really improves readability :-) (And I don't care about choice at all :-). Since it wasn't my code to start with, I'll leave it as is.

Reply:
Fair